lib/IRGen/GenFunc.cpp

```diff
@@ -2173,8 +2180,19 @@ Optional<StackAddress> irgen::emitFunctionPartialApplication(
                                                               subs,
                                                               &layout,
                                                               argConventions);
-  forwarder = emitPointerAuthSign(IGF, forwarder, outAuthInfo);
-  forwarder = IGF.Builder.CreateBitCast(forwarder, IGF.IGM.Int8PtrTy);
+  llvm::Value *forwarder;
+  if (origType->isAsync()) {
+    // TODO: Rather than creating an AsyncFunctionPointer in this case, use a
+    //       bare function pointer and teach apply/partial_apply lowering to
+    //       handle thick async functions differently. 
+    //       [lowering_thick_async_functions_without_sizes]
```

Comment:
I think this will cause TBDGen verification errors because we take the visibility of the async function pointer global off the `SILFn` but the partial apply forwarding thunk is internal and the async function pointer for it should also be internal.

```
public func callAsyncClosure(_ closure : () async->()) async {                  
  await closure()                                                               
}                                                                               
                                                                                
public class SomeClass {}

public func useAsyncClosure(_ c: SomeClass) async {                             
  func closure() async {                                                        
    print(c)                                                                    
  }                                                                             
  await callAsyncClosure(closure)                                               
} 
```

I know this is ultimately going away but in the meantime I think you could work around this by passing a null SILFunction.

Reply:
Just updated the PR, and AsyncFunctionPointers for partial apply forwarders are no longer being emitted.